### DISCUSSION TITLE
Sample rate

### DISCUSSION THREAD

**Discussion by stnor** [open-telemetry/opentelemetry-java#3235] [Q&A] [answered]
I think I'm seeing 100% sample rate in my app...

I'm using:
`-javaagent:/opt/aws-opentelemetry-agent-1.1.0.jar
-Dotel.resource.attributes=service.name=xxx
-Dotel.instrumentation.jdbc-datasource.enabled=true
-Dotel.instrumentation.opentelemetry-annotations.enabled=true
-Dotel.traces.sampler.traceidratio=true
-Dotel.traces.sampler.arg=0.005`

I would expect a 1/200 sample rate with that config, but I'm pretty sure it's at 100%. Is the syntax correct?

Ideally I'd love to set a number of samples per time unit rather than a percentage.

**Comment by jkwatson** [comment, marked as answer]
hm. this isn't a thing: `-Dotel.traces.sampler.traceidratio` .

You want: `-Dotel.traces.sampler=traceidratio`

Hope that helps.

**Reply by stnor** [reply]
Ah, cool. Thanks!

Do I need to worry about this: https://github.com/open-telemetry/opentelemetry-java/pull/2432 ?

**Reply by jkwatson** [reply]
Nope. Using `-Dotel.traces.sample.arg=0.005` is the correct way to do it.

BTW, you *might* want to consider using `parentbased_traceidratio` so you keep traces that are being sampled upstream of you as well.

**Reply by stnor** [reply]
OK. cool. Thanks so much! You saved my evening! :heart:
